src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,441 @@
+  signingScheme: SigningScheme.EIP712,
+}
+
+beforeEach(() => {
+  fetchMock.resetMocks()
+})
+
+afterEach(() => {
+  jest.restoreAllMocks()
+})
+
+test('Valid: Get an order ', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(ORDER_RESPONSE), { status: HTTP_STATUS_OK })
+  const order = await cowSdk.cowApi.getOrder('string')
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith('https://api.cow.fi/rinkeby/api/v1/orders/string', {
+    body: undefined,
+    headers: {
+      'Content-Type': 'application/json',
+      'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+    },
+    method: 'GET',
+  })
+  expect(order?.uid).toEqual('string')
+})
+
+test('Invalid: Get an order ', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+
+  try {
+    await cowSdk.cowApi.getOrder('notValidOrderId')
+  } catch (e: unknown) {
+    const error = e as OperatorError
+    expect(error.message).toEqual('Token pair selected has insufficient liquidity')
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith('https://api.cow.fi/rinkeby/api/v1/orders/notValidOrderId', {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    })
+  }
+})
+
+test('Valid: Get last 5 orders for a given trader ', async () => {
+  const ORDERS_RESPONSE = Array(5).fill(ORDER_RESPONSE)
+  fetchMock.mockResponseOnce(JSON.stringify(ORDERS_RESPONSE), { status: HTTP_STATUS_OK })
+  const orders = await cowSdk.cowApi.getOrders({
+    owner: '0x00000000005ef87f8ca7014309ece7260bbcdaeb', // Trader
+    limit: 5,
+    offset: 0,
+  })
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/rinkeby/api/v1/account/0x00000000005ef87f8ca7014309ece7260bbcdaeb/orders/?limit=5',
+    {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    }
+  )
+  expect(orders.length).toEqual(5)
+})
+
+test('Invalid: Get last 5 orders for an unexisting trader ', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+
+  try {
+    await cowSdk.cowApi.getOrders({
+      owner: 'invalidOwner',
+      limit: 5,
+      offset: 0,
+    })
+  } catch (e: unknown) {
+    const error = e as OperatorError
+    expect(error.message).toEqual('Token pair selected has insufficient liquidity')
```

Comment:
has this message something to do with the test? I mean, insufficient liquidity is about getting a price quote, but here we are querying a user's orders

Reply:
Yes, it is. Trying to getting orders from an invalid owner (i.e : malformed address) will return in a notFound error code which will be matched with that comment.